packages/fiori/src/NotificationAction.ts

```diff
@@ -0,0 +1,102 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import ButtonDesign from "@ui5/webcomponents/dist/types/ButtonDesign.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import Button from "@ui5/webcomponents/dist/Button.js";
+
+type NotificationActionClickEventDetail = {
+	targetRef: Button,
+};
+
+/**
+ * @class
+ * The <code>ui5-notification-action</code> represents an abstract action,
+ * used in the <code>ui5-li-notification</code> and the <code>ui5-li-notification-group</code> items.
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webc.fiori.NotificationAction
+ * @extends sap.ui.webc.base.UI5Element
+ * @abstract
+ * @tagname ui5-notification-action
+ * @implements sap.ui.webc.fiori.INotificationAction
+ * @public
+ */
+@event("click")
+@customElement("ui5-notification-action")
```

Comment:
let's reorder the decorators, we start with customElement as more important, and then the events

and 

let's fix something, that is not an issue with this PR, but in general, I noticed the "click" event is not shown in the API reference, and it's because it does not have the required JSDoc. And, also the detail is not described.
So, could you add the following as a comment for the "click" event + the detail info:


```js
/**
 * Fired, when the action is pressed.
 *
 * @event sap.ui.webc.fiori.NotificationAction#click
 * @param {HTMLElement} targetRef DOM ref of the clicked element
 * @public
 */
@event("click", {
	detail: {
		targetRef: { type: HTMLElement },
	},
})
```